### PR TITLE
Only add SubscriptionConfigurationFilePaths if UseFederatedAuth is true

### PR DESCRIPTION
**Pull request by danieljurek**
(and other test cleanup)

Fixed build (regular build): https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3912985&view=results

Fixed build (`UseFederatedAuth: true`): https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3912992&view=results -- in this case the failure happens in the [pre-script](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/test-resources-pre.ps1#L27). Unrelated to this change.

### CODE REVIEW

File: sdk/identity/tests.yml [hunk 1]
```diff
@@ -31,6 +31,6 @@ extends:
           # Contains alternate tenant, AAD app and cert info for testing
           - $(sub-config-identity-test-resources)
         ServiceConnection: azure-sdk-tests
-    EnvVars:
```
Comment (benbp) [inline sdk/identity/tests.yml:34]: Don't we still want the access token, or is that plumbed through underneath now?

**Review by benbp — APPROVED** [review]

File: sdk/identity/tests.yml (hunk 1 above)
Comment (danieljurek) [reply sdk/identity/tests.yml:34]: It's plumbed through underneath now: https://github.com/Azure/azure-sdk-for-net/blob/main/eng/pipelines/templates/jobs/live.tests.yml#L207 

This gives the following error: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3911090&view=results

File: eng/pipelines/templates/jobs/live.tests.yml [hunk 2]
```diff
@@ -115,7 +115,9 @@ jobs:
           SubscriptionConfiguration: ${{ parameters.CloudConfig.SubscriptionConfiguration }}
           SubscriptionConfigurations: ${{ parameters.CloudConfig.SubscriptionConfigurations }}
           EnvVars: ${{ parameters.EnvVars }}
-          SubscriptionConfigurationFilePaths: ${{ parameters.CloudConfig.SubscriptionConfigurationFilePaths }}
+          ${{ if parameters.UseFederatedAuth }}:
+            SubscriptionConfigurationFilePaths: ${{ parameters.CloudConfig.SubscriptionConfigurationFilePaths }}
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/live.tests.yml:119]: While this might fix the issue for now we probably need to fix the common script to handle null.

Comment (danieljurek) [reply eng/pipelines/templates/jobs/live.tests.yml:119]: https://github.com/Azure/azure-sdk-tools/issues/8507

**Review by weshaggard — APPROVED** [review]

**Comment by danieljurek** [comment]
@schaabs this fixes the issue discussed in chat https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3913365&view=logs&jobId=f0ef8d08-1538-5bca-8d2c-fdcdf6063e0d&j=f0ef8d08-1538-5bca-8d2c-fdcdf6063e0d&t=c025bcdc-58a3-5e56-e822-c74dc51c62ac ... your approval (or someone else from Azure/azure-sdk-write-identity) is required to merge.

**Comment by danieljurek** [comment]
Nevermind. We can fix the identity issues in another PR. This fix for non-federated live tests is ready to merge.

**Pull request merged**